### PR TITLE
Exclude JPype 0.7.2 from CI; add python3-setuptools for TeamCity

### PR DESCRIPTION
**Pull request by khaeru**
Required to un-break builds failing with messages like [this](https://ene-builds.iiasa.ac.at/viewLog.html?buildTypeId=IxmpCore_IxmpPython&buildId=2332&tab=buildLog&logTab=tree&filter=debug&expand=all&_focus=853).

The Dockerfile is also formatted identically to message_data.

## How to review

Confirm the TeamCity CI check passes.

## PR checklist
- [x] ~Tests added.~ N/A; CI only.
- [x] ~Documentation added.~ N/A
- [x] ~Release notes updated.~ N/A

### CODE REVIEW

**Comment by khaeru** [comment]
Note that a bug should be filed upstream with JPype to address the segfaults caused by 0.7.2.

File: ci/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+FROM openjdk:11-slim-buster
+
+# Install Python and other dependencies
+ARG VIRTUAL_ENV=/opt/python3
+
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends \
```
Comment (zikolach) [inline ci/Dockerfile]: nit: I would keep `&& \` at the end of line for consistency with other dockerfiles (other repos, see original one)

Comment (khaeru) [reply ci/Dockerfile]: Done.

**Comment by khaeru** [comment]
> better to use `git mv` (or similar from GUI tools) to be able to compare changes.

I did, but if the file has a high enough fraction of modified lines, then git (diff) views it as delete-and-create rather than rename-and-edit.

**Comment by khaeru** [comment]
Conda first installs JPype1 0.7.2, and then the ixmp dependency forces it back down to 0.7.1. For some reason, _only_ on AppVeyor/Python 3.7 (i.e. not 3.6), this caused [errors like](https://ci.appveyor.com/project/iiasa/ixmp/builds/31390731/job/48e6o00278femgvj):
```
___ ERROR at setup of TestTimeSeries.test_remove_geodata[TimeSeries-single] ___
request = <SubRequest 'test_mp' for <Function test_init[TimeSeries]>>
tmp_env = environ({'7ZIP': '"C:\\Program Files\\7-Zip\\7z.exe"', 'ALLUSERSPROFILE': 'C:\\ProgramData', 'APPDATA': 'C:\\Users\\ap...T_CURRENT_TEST': 'ixmp/tests/core/test_timeseries.py::TestTimeSeries::test_remove_geodata[TimeSeries-single] (setup)'})
test_data_path = WindowsPath('C:/projects/ixmp/ixmp/tests/data')
    @pytest.fixture(scope='class')
    def test_mp(request, tmp_env, test_data_path):
        """An empty ixmp.Platform connected to a temporary, in-memory database."""
        # Long, unique name for the platform.
        # Remove '/' so that the name can be used in URL tests.
        platform_name = request.node.nodeid.replace('/', ' ')
    
        # Add a platform
        ixmp_config.add_platform(platform_name, 'jdbc', 'hsqldb',
                                 url=f'jdbc:hsqldb:mem:{platform_name}')
    
        # Launch Platform
>       yield Platform(name=platform_name)
..\..\ixmp\testing.py:119: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\ixmp\core.py:86: in __init__
    self._backend = backend_class(**kwargs)
..\..\ixmp\backend\jdbc.py:182: in __init__
    properties = _create_properties(**kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
driver = 'hsqldb', path = None
url = 'jdbc:hsqldb:mem:ixmp tests core test_timeseries.py::TestTimeSeries'
user = None, password = None
    def _create_properties(driver=None, path=None, url=None, user=None,
                           password=None):
        """Create a database Properties from arguments."""
>       properties = java.Properties()
E       AttributeError: 'types.SimpleNamespace' object has no attribute 'Properties'
..\..\ixmp\backend\jdbc.py:64: AttributeError
```

This was resolved by also excluding the version in ci/conda-requirements.txt

**Pull request merged**